plugins/woocommerce/src/Admin/API/Reports/Products/DataStore.php

```diff
@@ -433,17 +467,106 @@ public static function sync_order_products( $order_id ) {
 
+		$is_full_refund_without_line_items = false;
+		$partial_refund_product_revenue    = array();
+		$refund_type                       = $order->get_meta( '_refund_type' );
+
+		$parent_order = null;
+
+		// When changing the order status to "Refunded", the refund order's type will be full refund, and the order items will be empty.
+		// We need to get the parent order items, and exclude the items that is already being patially refunded.
+		if (
+			'shop_order_refund' === $order->get_type() &&
+			'full' === $refund_type &&
+			empty( $order_items )
+		) {
+			$is_full_refund_without_line_items = true;
+
+			$parent_order_id = $order->get_parent_id();
+			$parent_order    = wc_get_order( $parent_order_id );
+			$order_items     = $parent_order->get_items();
+
+			// Get the partially refunded product and variation IDs along with their sum of product_net_revenue from the parent order.
+			$partial_refund_products = $wpdb->get_results(
+				$wpdb->prepare(
+					"
+						SELECT
+							product_lookup.product_id,
+							product_lookup.variation_id,
+							SUM( product_lookup.product_net_revenue ) AS product_net_revenue
+						FROM %i AS product_lookup
+						INNER JOIN {$wpdb->prefix}wc_order_stats AS order_stats
+							ON order_stats.order_id = product_lookup.order_id
+						WHERE 1 = 1
+							AND order_stats.parent_id = %d
+							AND product_lookup.product_net_revenue < 0
+						GROUP BY product_lookup.product_id, product_lookup.variation_id
+					",
+					$table_name,
+					$parent_order_id
+				)
+			);
+
+			/**
+			 * Create a lookup table for partially refunded products.
+			 * E.g. [
+			 *   '1' => -20,
+			 *   '2' => -40,
+			 *   '51' => -10,
+			 *   '52' => -30,
+			 * ]
+			 */
+			foreach ( $partial_refund_products as $product ) {
+				$id                                    = $product->variation_id ? $product->variation_id : $product->product_id;
+				$partial_refund_product_revenue[ $id ] = (float) $product->product_net_revenue;
+			}
+		}
+
 		foreach ( $order_items as $order_item ) {
 			$order_item_id = $order_item->get_id();
 			unset( $existing_items[ $order_item_id ] );
 			$product_qty         = $order_item->get_quantity( 'edit' );
+			$product_id          = $order_item->get_product_id( 'edit' );
+			$variation_id        = $order_item->get_variation_id( 'edit' );
 			$shipping_amount     = $order->get_item_shipping_amount( $order_item );
 			$shipping_tax_amount = $order->get_item_shipping_tax_amount( $order_item );
 			$coupon_amount       = $order->get_item_coupon_amount( $order_item );
+			$net_revenue         = round( $order_item->get_total( 'edit' ), $decimals );
+
+			// If the order is a full refund and there is no order items. The order item here is the parent order item.
+			if ( $is_full_refund_without_line_items ) {
+				$id             = $variation_id ? $variation_id : $product_id;
+				$partial_refund = $partial_refund_product_revenue[ $id ];
```

Comment:
This needs a fallback to set to 0 as there might not be a partial refund for this product. It's causing the unit tests to fail.

This was my previous suggestion:
```suggestion
				$partial_refund = $partial_refund_product_revenue[ $id ] ?? 0;
```